aws/resource_aws_lambda_alias.go

```diff
@@ -97,6 +126,7 @@ func resourceAwsLambdaAliasRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("function_version", aliasConfiguration.FunctionVersion)
 	d.Set("name", aliasConfiguration.Name)
 	d.Set("arn", aliasConfiguration.AliasArn)
+	d.Set("routing_config", aliasConfiguration.RoutingConfig)
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
if err := d.Set("routing_config", aliasConfiguration.RoutingConfig); err != nil {
  return fmt.Errorf("error setting routing_config: %s", err)
}
```

Without this check and if there is a type conversion error, Terraform is silently accepting the configuration "as-is" into the Terraform state and unable to detect drift from the API should changes occur outside Terraform.

After adding this error checking, you should start receiving type errors as `aliasConfiguration.RoutingConfig` is a SDK object and not a `[]interface{}`. We generally handle this via a "flattenServiceObject()" function. e.g.

```go

func flattenLambdaAliasRoutingConfiguration(arc *lambda.AliasRoutingConfiguration) []interface{} {
	if arc == nil {
		return []interface{}{}
	}

	m := map[string]interface{}{
		"additional_version_weights": aws.Float64ValueMap(arc.AdditionalVersionWeights)
	}

	return []interface{}{m}
}
```

Which can be referenced via:

```
if err := d.Set("routing_config", flattenLambdaAliasRoutingConfiguration(aliasConfiguration.RoutingConfig)); err != nil {
  return fmt.Errorf("error setting routing_config: %s", err)
}
```